core/aggregatedpropertymodel.cpp

```diff
@@ -475,22 +477,23 @@ bool AggregatedPropertyModel::hasLoop(PropertyAdaptor *adaptor, const QVariant &
 void AggregatedPropertyModel::reloadSubTree(PropertyAdaptor *parentAdaptor, int index)
 {
     Q_ASSERT(parentAdaptor);
-    Q_ASSERT(m_parentChildrenMap.contains(parentAdaptor));
+    Q_ASSERT(m_parentChildrenMap.find(parentAdaptor) != m_parentChildrenMap.cend());
     Q_ASSERT(index >= 0);
-    Q_ASSERT(index < m_parentChildrenMap.value(parentAdaptor).size());
+    Q_ASSERT(index < m_parentChildrenMap.at(parentAdaptor).size());
 
     // prevent rowCount calls as a result of the change notification to re-create
     // the adaptor
     m_inhibitAdaptorCreation = true;
 
     // remove the old sub-tree, if present
-    auto oldAdaptor = m_parentChildrenMap.value(parentAdaptor).at(index);
+    auto oldAdaptor = m_parentChildrenMap.at(parentAdaptor).at(index);
     if (oldAdaptor) {
-        auto oldRowCount = m_parentChildrenMap.value(oldAdaptor).size();
+        auto oldRowCount = m_parentChildrenMap.at(oldAdaptor).size();
         if (oldRowCount > 0)
             beginRemoveRows(createIndex(index, 0, parentAdaptor), 0, oldRowCount - 1);
         m_parentChildrenMap[parentAdaptor][index] = nullptr;
-        m_parentChildrenMap.remove(oldAdaptor);
+
+        m_parentChildrenMap.erase(m_parentChildrenMap.find(oldAdaptor));
```

Comment:
double lookup not needed I think, iterator from the line `auto oldRowCount = m_parentChildrenMap.at(oldAdaptor).size();` can be used here.

Reply:
Fixed